src/main/java/ui/components/pickers/MilestonePickerState.java

```diff
@@ -30,13 +30,23 @@ public MilestonePickerState(List<PickerMilestone> milestones, String userInput)
     }
 
     private void processInput(String userInput) {
+        filterMilestonesByInput(currentMilestones, userInput);
+        determineBestMatchingMilestones(userInput);
         if (userInput.isEmpty()) return;
+        toggleFirstMilestone(bestMatchingMilestones);
+    }
 
+    private void filterMilestonesByInput(List<PickerMilestone> milestones, String userInput) {
         String[] userInputWords = userInput.split(" ");
         Stream.of(userInputWords)
-                .forEach(this::filterMilestones);
+                .forEach(query -> filterMilestonesByQuery(milestones, query));
+    }
 
-        toggleFirstMatchingMilestone(currentMilestonesList);
+    private void filterMilestonesByQuery(List<PickerMilestone> milestones, String query) {
+        milestones.stream()
+                .forEach(milestone -> {
+                    if (milestone.isMatching()) milestone.setMatching(isMatchingQuery(milestone, query));
+                });
     }
```

Comment:
The difference between this method and the one above is the number of words in the query? In that case names can reflect that.
e.g. `filterMilestonesByQuerySentence` `filterMilestonesByQueryWord`